[MM-48947] app/platform: do not restart metrics on every config save

isacikgoz opened the pull request:
#### Summary
It seems like there are no checks if a metrics server restart is required or not.

#### Ticket Link
https://mattermost.atlassian.net/browse/MM-48947

#### Release Note

```release-note
Bugfix on metrics server restart when there is a config change.
```

isacikgoz commented on app/platform/service_test.go:
```diff
@@ -117,17 +116,14 @@ func TestMetrics(t *testing.T) {
 	})
 
 	t.Run("ensure the metrics server is started with advanced metrics", func(t *testing.T) {
-		t.Skip("MM-47635")
```
This must be creep in by https://github.com/mattermost/mattermost-server/pull/21401

agarciamontoro reviewed (approved):
Neat, thanks!

isacikgoz commented:
/e2e-test

mattermod commented:
Successfully triggered E2E testing!
[GitLab pipeline](https://git.internal.mattermost.com/qa/cypress-ui-automation/-/pipelines/273272) | [Test dashboard](https://automation-dashboard.vercel.app/cycle/273272)

isacikgoz commented:
@agnivade requesting a review again as I removed the trigger from `SaveConfig` and added to config listeners. Because there are cases where the config is getting updated with `UpdateConfig` method and this trigger was getting ignored.

amyblais commented:
/cherry-pick cloud

mattermod commented:
Cherry pick is scheduled.
